src/ray/util/tests/pipe_logger_test.cc

```diff
@@ -41,89 +44,104 @@ TEST_P(PipeLoggerTest, NoPipeWrite) {
          absl::StrFormat("%d", pipe_buffer_size).data(),
          /*overwrite=*/1);
 
-  // TODO(core): We should have a better test util, which allows us to create a temporary
-  // testing directory.
+  // TODO(hjiang): We should have a better test util, which allows us to create a
+  // temporary testing directory.
   const std::string test_file_path = absl::StrFormat("%s.out", GenerateUUIDV4());
 
-  // Take the default option, which doesn't have rotation enabled.
-  StreamRedirectionOption logging_option{};
-  logging_option.file_path = test_file_path;
-  auto log_token = CreateRedirectionFileHandle(logging_option);
+  // Delete temporary file.
+  absl::Cleanup cleanup_test_file = [&test_file_path]() {
+    EXPECT_TRUE(std::filesystem::remove(test_file_path));
+  };
 
-  ASSERT_EQ(write(log_token.GetWriteHandle(), kLogLine1.data(), kLogLine1.length()),
-            kLogLine1.length());
-  ASSERT_EQ(write(log_token.GetWriteHandle(), kLogLine2.data(), kLogLine2.length()),
-            kLogLine2.length());
-  log_token.Close();
+  // Take the default option, which doesn't have rotation enabled.
+  StreamRedirectionOption stream_redirection_opt{};
+  stream_redirection_opt.file_path = test_file_path;
+  auto stream_redirection_handle = CreateRedirectionFileHandle(stream_redirection_opt);
+  stream_redirection_handle.CompleteWrite(kLogLine1.data(), kLogLine1.length());
+  stream_redirection_handle.CompleteWrite(kLogLine2.data(), kLogLine2.length());
+  stream_redirection_handle.Close();
 
   // Check log content after completion.
   const auto actual_content = CompleteReadFile(test_file_path);
   const std::string expected_content = absl::StrFormat("%s%s", kLogLine1, kLogLine2);
   EXPECT_EQ(actual_content, expected_content);
-
-  // Delete temporary file.
-  EXPECT_EQ(unlink(test_file_path.data()), 0);
 }
 
+INSTANTIATE_TEST_SUITE_P(PipeLoggerTest, PipeLoggerTest, testing::Values(1024, 3));
+
+}  // namespace
+
+}  // namespace ray
+
+/////////////////////////////////////////////////
+// Unit test for both unix platform only.
+/////////////////////////////////////////////////
+
+#if defined(__APPLE__) || defined(__linux__)
+
+#include <unistd.h>
+
+namespace ray {
+
+namespace {
+
 TEST_P(PipeLoggerTest, PipeWrite) {
```

Comment:
Not sure whether if it will work given PipeLoggerTest is instantiated in a different anonymous namespace but if it works, I'm fine with it.

Reply:
all tests should be placed under anonymous namespace to prevent symbol leak; another option is visibility compilation option, which is another option

Reply:
I'm not saying to not put it under anonymous namespace. Anyway, if it works, we can merge it.